### PR TITLE
intial tests

### PR DESCRIPTION
**Pull request by jsirish**
* travis updates for SilverStripe 4
* scrutinizer updates
* codecov setup

### CODE REVIEW

**Comment by jsirish** [comment]
@sheadawson when you have a chance, could you turn on Travis for the repo? I can re-push the PR afterwards to trigger the tests.

**Comment by sheadawson** [comment]
@jsirish I've activated the repo in travis. Let me know if you need anything else

**Comment by jsirish** [comment]
@sheadawson awesome thanks!

File: tests/BlockSetTest.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                'name' => _t('BlockSet.DeleteBlockSet','Delete a Block Set'),
+                'category' => _t('Block.PermissionCategory', 'Blocks'),
+            ],
+            'BLOCKSET_CREATE' => [
+                'name' => _t('BlockSet.CreateBlockSet','Create a Block Set'),
+                'category' => _t('Block.PermissionCategory', 'Blocks'),
+            ],
+        ];
+
+        $this->assertEquals($object->providePermissions(), $expected);
```
Comment (muskie9) [inline tests/BlockSetTest.php]: The only thing here is expected generally comes first, so just switching the two parameters.

File: tests/BlockTest.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+                'name' => _t('Block.CreateBlock', 'Create a Block'),
+                'category' => _t('Block.PermissionCategory', 'Blocks'),
+            ],
+            'BLOCK_PUBLISH' => [
+                'name' => _t('Block.PublishBlock', 'Publish a Block'),
+                'category' => _t('Block.PermissionCategory', 'Blocks'),
+            ],
+        ];
+
+        $this->assertEquals($object->providePermissions(), $expected);
```
Comment (muskie9) [inline tests/BlockTest.php]: The only thing here is expected generally comes first, so just switching the two parameters.

**Pull request merged**